### DISCUSSION TITLE
Layout Component Class is ignored for full-page Livewire Components

### DISCUSSION THREAD

**Discussion by selfsimilar** [livewire/livewire#4595] [7. Bugs (archived)] [answered]
It appears that if a Livewire component is instantiated as a full-page component, the renderer ignores the layout component's class file.

To Reproduce:
In a fresh install of Laravel + livewire, run

    $ php artisan make:livewire FullPage
    $ php artisan make:controller ExampleController
    $ touch resources/views/example.blade.php

add the following routes:

    use App\Http\Controllers\ExampleController;
    use App\Http\Livewire\FullPage;
    [...]
    Route::get('livewire-embedded', [ExampleController::class, 'index']);
    Route::get('livewire-fullpage', FullPage::class);

add to `app/Http/Livewire/FullPage.php`

    public function render()
    {
        return view('livewire.full-page')
            ->layout('layouts.guest');
    }

add to `app/Http/Controllers/ExampleController.php`

    public function index()
    {
        return view('example');
    }

add to `resources/views/example.blade.php`

    <x-guest-layout>
        <livewire:full-page />
    </x-guest-layout>

At this point, both routes work and show whatever is in the Livewire component.

Now, add the following to `app/Views/Components/GuestLayout.php`

    public $test = 'something';
    
    public function __construct()
    {
        $this->test = 'something else';
    }

and edit `resources/views/layouts/guest.blade.php` to display the contents of `$test`.

The route `livewire-embedded` will still work and can show `$test` as 'something else', but `livewire-fullpage` will throw the error `Undefined variable $test`.

**Comment by joshhanley** [comment, marked as answer]
@selfsimilar I'm pretty sure support for that was added not to long ago. But if I remember rightly, I think you need to pass the layout php class in, instead of the view name. Like:

```php
public function render()
{
    return view('livewire.full-page')
        ->layout(\App\Views\Components\GuestLayout::class);
}
```

Give that a try and let us know how it goes.

Hope this helps!

**Reply by selfsimilar** [reply]
Thanks, this worked. I'll submit a PR to the Livewire docs to make this clearer.

**Reply by joshhanley** [reply]
Thanks!
